core/src/main/java/com/netflix/conductor/service/Lifecycle.java

```
@@ -0,0 +1,18 @@
+package com.netflix.conductor.service;
+
+public interface Lifecycle {
+
+    default void start() throws Exception {
+        registerShutdownHook();
+    }
+
+    void stop() throws Exception;
+
+    default void registerShutdownHook() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                stop();
+            } catch (Exception e) {}
```

don't swallow exception